src/libraries/System.Private.CoreLib/src/System/Globalization/SortKey.cs

```diff
@@ -62,44 +62,34 @@ public static int Compare(SortKey sortkey1, SortKey sortkey2)
             Debug.Assert(key1Data != null, "key1Data != null");
             Debug.Assert(key2Data != null, "key2Data != null");
 
-            if (key1Data.Length == 0)
-            {
-                if (key2Data.Length == 0)
-                {
-                    return 0;
-                }
+            // SortKey comparisons are done as an ordinal comparison by the raw sort key bytes.
```

Comment:
`return new ReadOnlySpan<byte>(key1Data).SequenceCompareTo(keyData2);` ?

Reply:
I can't believe I'd forgotten about that API. 😳